triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Server.scala

```diff
@@ -338,16 +339,18 @@ class Server(
                   // TODO[AH] Why do we need to pass ec, system explicitly?
                   authorize(claims)(ec, system) {
                     token =>
-                      val instOrErr =
+                      val instOrErr: Future[Either[String, JsValue]] =
                         addNewTrigger(params.party, params.triggerName, params.applicationId, token)
-                          .map {
+                          .map(_.map {
                             case (trigger, runningTrigger) => startTrigger(trigger, runningTrigger)
-                          }
-                      instOrErr match {
-                        case Left(err) =>
+                          })
+                      onComplete(instOrErr) {
+                        case Failure(exception) =>
+                          complete(
+                            errorResponse(StatusCodes.InternalServerError, exception.description))
```

Comment:
Returning arbitrary exceptions to the client is maybe something we should avoid. Not to risk exposing any sensitive information about a deployment. It might be better to log the error and return a generic error message. For certain types of errors we can choose to be more specific, e.g. unprocessable entity. I think this was already an issue before this PR, so seems more appropriate for a follow up PR.

Reply:
Yeah that’s a good point, agree on tackling it separately.